### PR TITLE
Removes junit 4 dependency

### PR DESCRIPTION
**Pull request by codefromthecrypt**
This changes to the next minor so that we can remove the junit 4 compile dependency. To do so, this ports all tests to jupiter (v5) and stops publishing zipkin-junit (v4).

The plan is to cut a new version without any changes to the main tree. This will allow a cleaner cutover for server extensions who can first move to the latest patch, and subsequently the next minor. Specifically, this is for zipkin-gcp and zipkin-aws and I personally will handle this.

Note: Certain test libraries, notably testcontainers and activemq-junit have an indirect dependency on old junit. However, we don't re-export that in our zipkin-tests or zipkin-junit5 modules.

While I had to do some manual work, I had a lot of help from [openrewrite](https://junit.org/junit5/docs/current/user-guide/#migrating-from-junit4). Here's the part that helped a lot.

```bash
$ ./mvnw -U org.openrewrite.maven:rewrite-maven-plugin:run \
  -Drewrite.recipeArtifactCoordinates=org.openrewrite.recipe:rewrite-testing-frameworks:LATEST \
  -Drewrite.activeRecipes=org.openrewrite.java.testing.junit5.JUnit4to5Migration
````

### CODE REVIEW

**Comment by codefromthecrypt** [comment]
draft until green

**Comment by codefromthecrypt** [comment]
manually verified also, since there are a lot of lines.

```bash
$ ag 'import .*junit.' --java --noheading |grep -v '^\s*$' | wc -l
     352
$ ag 'import .*junit.' --java --noheading |egrep -v '(^\s*$|junit5|jupiter)'
zipkin-collector/activemq/src/test/java/zipkin2/collector/activemq/ITActiveMQCollector.java:27:import org.apache.activemq.junit.EmbeddedActiveMQBroker;
```

File: pom.xml [hunk 1]
```diff
@@ -187,37 +185,14 @@
     <!-- Do not add compile dependencies here. This can cause problems for libraries that depend on
          io.zipkin.zipkin2:zipkin difficult to unravel. -->
 
-    <dependency>
-      <groupId>junit</groupId>
```
Comment (anuraaga) [inline pom.xml:191]: Skimmed through a few files for kicks but this is probably the main change and if build is green then it looks like it worked well

**Review by anuraaga — APPROVED** [review]
.

**Comment by timtebeek** [comment]
Might be worth to followup with these AssertJ best practices once this is merged, or cherry pick that into here:
https://github.com/openzipkin/zipkin/compare/master...timtebeek:zipkin:refactor/assertj-best-practices

**Comment by codefromthecrypt** [comment]
Thanks @timtebeek I'll create a PR off your changes next.

**Comment by timtebeek** [comment]
I'm seeing there's conflicts; would you want me to recreate the changes? That's trivial with the platform we have.

**Comment by codefromthecrypt** [comment]
added https://github.com/openzipkin/zipkin/pull/3630 and will cut the minor afterwards

**Pull request merged**